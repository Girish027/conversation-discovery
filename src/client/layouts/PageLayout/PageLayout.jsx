import React, { PureComponent } from 'react';
import NavBar from '../../components/NavBar';
import PageHeader from '../../components/PageHeader';
import HomePage from '../../components/HomePage';
import ConnectedModal from '../../components/Modals/Modal';
import '../../styles/layout.scss';

export class PageLayout extends PureComponent {
  render() {
    return (
      <div className='container'>
        <header>
          <NavBar />
          <PageHeader />
        </header>
        <HomePage />
        <ConnectedModal />
        <footer></footer>
      </div>
    );
  }
}

export default PageLayout;
