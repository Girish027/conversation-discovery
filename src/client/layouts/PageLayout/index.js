import ConnectedPageLayout from './PageLayout';

export default ConnectedPageLayout;
