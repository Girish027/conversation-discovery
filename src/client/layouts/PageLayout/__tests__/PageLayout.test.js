import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { PageLayout } from 'layouts/PageLayout/PageLayout';

describe('<PageLayout />', function () {
  let wrapper;

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getShallowWrapper = (props = {}) => shallow(
    <PageLayout {...props} />
  );

  describe('Creating an instance', () => {
    test('should exist - connected component PageLayout', () => {
      wrapper = getShallowWrapper();
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly with all the necessary components', () => {
      wrapper = getShallowWrapper();
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
