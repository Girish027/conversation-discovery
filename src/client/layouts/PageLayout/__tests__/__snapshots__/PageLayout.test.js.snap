// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLayout /> Snapshots renders correctly with all the necessary components 1`] = `
<div
  className="container"
>
  <header>
    <Connect(withTheme(NavBar)) />
    <Connect(PageHeaderWrapper) />
  </header>
  <HomePage />
  <Connect(Modal) />
  <footer />
</div>
`;
